File: Solutions/221. Maximal Square/221. Maximal Square.cpp
class Solution {
public:
    int maximalSquare(vector<vector<char>>& mat) {
        int m=mat.size(),n=mat[0].size();
        int edge=0;
        vector<vector<int>> dp(m,vector<int>(n));
        for(int i=0;i<m;++i)
            for(int j=0;j<n;++j){
                if(i>0&&j>0&&mat[i][j]=='1')
                    dp[i][j]=min({dp[i-1][j-1],dp[i][j-1],dp[i-1][j]})+1;
                else
                    dp[i][j]=mat[i][j]=='1'?1:0;
                edge=max(edge,dp[i][j]);
            }
        return pow(edge,2);
    }
};

File: Solutions/221. Maximal Square/221. Maximal Square.py
class Solution:
    def maximalSquare(self, m: List[List[str]]) -> int:
        edge,dp=0,[[0]*len(m[0]) for _ in range(len(m))]
        for i in range(len(m)):
            for j in range(len(m[0])):
                if m[i][j]=='1' and i>0 and j>0:
                    dp[i][j]=1+min(dp[i-1][j-1],dp[i][j-1],dp[i-1][j])
                else:
                    dp[i][j]=int(m[i][j])
                edge=max(edge,dp[i][j])
        return edge*edge